cdap-ui/cypress/integration/pipeline.joiner.spec.ts

```diff
@@ -0,0 +1,175 @@
+  });
+
+  after(() => {
+    // Delete the pipeline to clean up
+    cy.cleanup_pipelines(headers, JOINER_PIPELINE_NAME);
+  });
+
+  it('Should be able to upload a pipeline with joiner widget', () => {
+    cy.visit('/pipelines/ns/default/studio');
+    cy.get('#resource-center-btn').click();
+    cy.get('#create-pipeline-link').click();
+    cy.url().should('include', '/studio');
+
+    cy.upload_pipeline(
+      'joiner_pipeline_v1-cdap-data-pipeline.json',
+      '#pipeline-import-config-link > input[type="file"]'
+    );
+    // This is arbitrary. Right now we don't have a way to determine
+    // if the upgrade check is done. Since this a standalone the assumption
+    // is this won't take more than 10 seconds.
+    cy.wait(10000);
+    // Name pipeline
+    cy.get('.pipeline-name').click();
+    cy.get('#pipeline-name-input')
+      .clear()
+      .type(JOINER_PIPELINE_NAME)
+      .type('{enter}');
+  });
+
+  it('Should show appropriate message when preview has not been run yet', () => {
+    cy.window().then((window) => {
+      skipPreviewTests = window.CDAP_CONFIG.hydrator.previewEnabled !== true;
+    });
+    if (skipPreviewTests) {
+      skip();
+    }
+    cy.get(dataCy('pipeline-preview-btn')).click();
+    cy.get(dataCy(`${joinerNode.nodeName}-preview-data-btn`)).click();
+    cy.contains(runPreviewMsg).should('be.visible');
+    cy.get(closeButton).click();
+  });
+
+  it('Should show appropriate message when preview has been stopped before data is generated', () => {
+    cy.window().then((window) => {
+      skipPreviewTests = window.CDAP_CONFIG.hydrator.previewEnabled !== true;
+    });
+    if (skipPreviewTests) {
+      skip();
+    }
+    // Start and then immediately stop preview
+    cy.get(dataCy('run-preview-btn')).click();
+    cy.get(dataCy('stop-preview-btn')).click();
+    cy.get(dataCy('run-preview-btn'), { timeout: 30000 }).should('exist');
+    cy.get(
+      dataCy(`plugin-node-${sourceNode1.nodeName}-${sourceNode1.nodeType}-${sourceNode1.nodeId}`)
+    ).within(() => {
+      cy.get(dataCy(`${sourceNode1.nodeName}-preview-data-btn`)).click();
+    });
+    cy.contains(noDataMsg).should('be.visible');
+    cy.get(closeButton).click();
+  });
+
+  it('Should show preview data with table view by default for sink', () => {
+    cy.window().then((window) => {
+      skipPreviewTests = window.CDAP_CONFIG.hydrator.previewEnabled !== true;
+    });
+    if (skipPreviewTests) {
+      skip();
+    }
+    // Start and run preview
+    cy.get(dataCy('run-preview-btn')).click();
+    cy.get(dataCy('stop-preview-btn')).should('be.visible');
+    cy.get(dataCy('run-preview-btn'), { timeout: 70000 }).should('exist');
+    cy.get(dataCy(`plugin-node-TPFSAvro-batchsink-2`)).within(() => {
+      cy.get(dataCy(`${sinkNode.nodeName}-preview-data-btn`)).click();
+    });
+    // Should be able to toggle view and navigate records
+    cy.get(dataCy('toggle-Table'), { timeout: 10000 }).should('exist');
```

Comment:
is there any reason why Table is captialized?

Reply:
That part of the attribute comes directly from the user-facing labels on the ToggleSwitch, which are capitalized and come from the PreviewMode enum (https://github.com/cdapio/cdap/blob/0a332be257169d1ed0e2fd046ca03d5db34791b5/cdap-ui/app/cdap/components/PreviewData/index.tsx#L62).